### DISCUSSION TITLE
OSD for the hardware controls

### DISCUSSION THREAD

**Discussion by rp1231** [File-New-Project/EarTrumpet#740] [Ideas]
Would it be possible to have a volume mixer OSD popup every time you change volume?
I struggle to know the volume changes when I use hardware controls and an OSD would be perfect!

Thanks

**Comment by rp1231** [comment]
Any interest for this?
This would go well with the hardware support addon being developed.

**Comment by rp1231** [comment]
Maybe it can be as simple as having the eartrumpet app popping up every time an app volume is changed via hardware controls.
![EarTrumpet_AMR3lsPGGX](https://user-images.githubusercontent.com/47380919/108110834-dcc39d00-70b9-11eb-8785-876a70c89895.png)
Or just the individual app showing up.
![Discord](https://user-images.githubusercontent.com/47380919/108110888-f06f0380-70b9-11eb-8c43-f8878efc103e.png)

I could help with the design assets if needed.
Thanks

**Comment by svenwml** [comment]
In my opinion, this would be a great addition, e.g. when using the hardware support addon with rotary encoders which do not allow to tell the absolute position. But I think this would have to be part of the main application with a setting to enable / disable it.

**Reply by rp1231** [reply]
Ahh ok, makes sense!
I hope one of the devs working on the main application picks this up.

**Reply by davuxcom** [reply]
@svenwml I gave you access to the addons repo that will be public soon.

I've added a sample for adding an OSD: https://github.com/File-New-Project/EarTrumpet.Addons/commit/e2095ba64cdf9274531ecf6717a0ee4dfb1c0877

You'll also need the current `dev` branch so you have `DeviceView!IsAppListVisible`. Otherwise, there aren't any changes needed to have an OSD that displays whatever you like and doesn't conflict with the EarTrumpet flyout.

![image](https://user-images.githubusercontent.com/1456216/108290233-d62e3780-7144-11eb-8abc-79cc777e32d0.png)

![image](https://user-images.githubusercontent.com/1456216/108290258-e219f980-7144-11eb-8498-c983975bcf84.png)

It'd probably be easier to show something totally custom, this is `DeviceView` and `AppItemView`, it's set to show on top but `IsHitTest=false` so you cannot interact with it. Ideally I think not showing the slider would be much better, but exactly what to show would be up to you...

**Reply by svenwml** [reply]
@davuxcom This looks great, thanks! We will definitely have a look at this and plan to implement the option to show an OSD on hardware control change in the hardware controls addon.

**Comment by LifeJustDLC** [comment]
Any progress on this? If this is not planned, I will consider make a browser extension instead.
